Refactor build script

munechika-koyo opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

munechika-koyo committed:
MNT: Update build script to use dynamic package name for installation

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

munechika-koyo committed:
MNT: Re-rendered with conda-smithy 3.52.2 and conda-forge-pinning 2025.09.25.19.16.51

Other tools:
- conda-build 25.7.0
- rattler-build 0.47.0
- rattler-build-conda-compat 1.4.6

munechika-koyo committed:
MNT: Increment build number from 0 to 1 in recipe.yaml

munechika-koyo commented:
@conda-forge-admin, please add bot automerge

munechika-koyo committed:
MNT: Fix comment and sed command for stripping buildtype from MESON_ARGS

munechika-koyo committed:
MNT: Update test section to include additional imports and simplify unittest command

munechika-koyo committed:
MNT: Re-rendered with conda-smithy 3.52.3 and conda-forge-pinning 2025.10.08.21.44.29

Other tools:
- conda-build 25.9.0
- rattler-build 0.47.1
- rattler-build-conda-compat 1.4.6
